op-deployer/pkg/deployer/pipeline/interop_depset.go

```diff
@@ -10,21 +10,29 @@ import (
 	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
 )
 
-func GenerateInteropDepset(ctx context.Context, pEnv *Env, globalIntent *state.Intent, st *state.State) error {
+func GenerateInteropDepset(_ context.Context, pEnv *Env, globalIntent *state.Intent, st *state.State) error {
 	lgr := pEnv.Logger.New("stage", "generate-interop-depset")
 
-	if !globalIntent.UseInterop {
-		lgr.Warn("interop not enabled - skipping interop depset generation")
-		return nil
-	}
-
 	lgr.Info("creating interop dependency set...")
 	deps := make(map[eth.ChainID]*depset.StaticConfigDependency)
 	for i, chain := range globalIntent.Chains {
 		id := eth.ChainIDFromBytes32(chain.ID)
+		_, config, err := calculateL2GenesisOverrides(globalIntent, chain)
+		if err != nil {
+			return fmt.Errorf("failed to calculate L2 genesis overrides for chain %v: %w", chain.ID, err)
+		}
+		if config.L2GenesisInteropTimeOffset == nil {
+			// Don't add chains to the dep set if they don't have interop scheduled.
+			continue
+		}
```

Comment:
We'll need to change this later, so we can have a depset config before interop is scheduled, to be able to sync with supervisor ahead of interop. But only after we simplify the config (as per discord thread).